plugin/uuid.go

```diff
@@ -0,0 +1,28 @@
+package plugin
+
+import (
+	"fmt"
+	"strconv"
+)
+
+const uuidPattern = "^([a-fA-F0-9]{8}-" +
+	"[a-fA-F0-9]{4}-" +
+	"[%s]" +
+	"[a-fA-F0-9]{3}-" +
+	"[8|9|aA|bB][a-fA-F0-9]{3}-" +
+	"[a-fA-F0-9]{12})?$"
```

Comment:
I am not entirely sure that we should be allowing for empty strings to pass the regex. If one sets a `uuid_ver` constraint on a field I expect that field to always be set.

For the case where the field is optional (and if set should be a UUID) we can add another boolean validator `uuid_empty` which if set allows for the field to be empty.

I haven't fully made up my mind up on whether this is the right approach so am open for discussion and suggestions here.

Reply:
Allowing empty strings by default is not straightforward for the user.
I agree that it would be better if it must be explicitly stated.

In general I like the idea to have combinable validators. 
If you want to fail validator if an uuid field is empty you can already pass the `string_not_empty` validator, like:
`{uuid_ver: 4, string_not_empty: true}`.

The opposite variant would be better for this usecase though:
`{uuid_ver: 4, allow_empty_string: true}`.